File: app/src/main/java/com/remotearthsolutions/expensetracker/utils/AppbrainAdUtils.java
package com.remotearthsolutions.expensetracker.utils;

import android.app.Activity;
import com.appbrain.AdId;
import com.appbrain.InterstitialBuilder;
import com.appbrain.InterstitialListener;
import com.remotearthsolutions.expensetracker.activities.ApplicationObject;

public class AppbrainAdUtils {

    private static AppbrainAdUtils instance;
    private Activity activity;
    private InterstitialBuilder interstitialBuilder;

    public static AppbrainAdUtils getInstance(Activity activity) {
        if (instance == null) {
            instance = new AppbrainAdUtils(activity);
        }

        return instance;
    }

    private AppbrainAdUtils(Activity activity) {
        this.activity = activity;
    }

    public void showAds() {

        interstitialBuilder = InterstitialBuilder.create()
                .setAdId(AdId.LEVEL_COMPLETE)
                .setListener(new InterstitialListener() {
                    @Override
                    public void onPresented() {

                    }

                    @Override
                    public void onClick() {

                    }

                    @Override
                    public void onDismissed(boolean b) {

                    }

                    @Override
                    public void onAdLoaded() {
                        ApplicationObject app = (ApplicationObject) activity.getApplication();
                        if (app.isActivityVisible() && app.isAppShouldShowAds()) {
                            interstitialBuilder.show(activity);
                            FabricAnswersUtils.logCustom("Appbrain Ad shown");
                        }
                    }

                    @Override
                    public void onAdFailedToLoad(InterstitialError interstitialError) {

                    }
                })
                .preload(activity);

    }
}

File: app/src/main/java/com/remotearthsolutions/expensetracker/utils/AppbrainAdUtils.kt
package com.remotearthsolutions.expensetracker.utils

import android.app.Activity
import com.appbrain.AdId
import com.appbrain.InterstitialBuilder
import com.appbrain.InterstitialListener
import com.appbrain.InterstitialListener.InterstitialError
import com.remotearthsolutions.expensetracker.activities.ApplicationObject
import com.remotearthsolutions.expensetracker.utils.FabricAnswersUtils.logCustom

class AppbrainAdUtils private constructor(private val activity: Activity) {
    private var interstitialBuilder: InterstitialBuilder? = null
    fun showAds() {
        interstitialBuilder = InterstitialBuilder.create()
            .setAdId(AdId.LEVEL_COMPLETE)
            .setListener(object : InterstitialListener {
                override fun onPresented() {}
                override fun onClick() {}
                override fun onDismissed(b: Boolean) {}
                override fun onAdLoaded() {
                    val app = activity.application as ApplicationObject
                    if (app.isActivityVisible && app.isAppShouldShowAds) {
                        interstitialBuilder!!.show(activity)
                        logCustom("Appbrain Ad shown")
                    }
                }

                override fun onAdFailedToLoad(interstitialError: InterstitialError) {}
            })
            .preload(activity)
    }

    companion object {
        private var instance: AppbrainAdUtils? = null
        fun getInstance(activity: Activity): AppbrainAdUtils? {
            if (instance == null) {
                instance =
                    AppbrainAdUtils(activity)
            }
            return instance
        }
    }

}